torch/csrc/inductor/aoti_torch/c/shim.h

```diff
@@ -74,6 +74,11 @@ using AOTITorchError = int32_t;
 #define AOTI_TORCH_SUCCESS 0
 #define AOTI_TORCH_FAILURE 1
 
+struct ShimOptional {
+  bool has_value;
+  int64_t value;
```

Comment:
This is not general to other types?

Reply:
It's general, I just picked a random 64-bit type. The actual type gets casted here: https://github.com/pytorch/pytorch/pull/112527/files#diff-b60511be1e7fafc2c45e7c0cb3e769ad48b2a1060a69759f58979ffc33b38a79R53